### PR TITLE
Draft: Classify spectra and refactor `mainHeader`

### PR DESCRIPTION
**Pull request by santimirandarp**
Hi,

TL;DR
* This is a draft, function is very rudimentary yet. Would be good to get your thoughts.
* I have added a basic [`guessType` function](https://github.com/santimirandarp/spc-parser/blob/0f2ac69fe7d0eb139d5d3223b9a22ad2f6621126/src/utility.ts#L104) and the tests.
* Created 2 classes `OldHeader`, `NewHeader` and used `instanceof` to fix any dependency
* Now `typedoc` generates documentation (it was generating only for `parse` function.)

----------

I bit long winded but I couldn't quite split the commits in this case.

From the SPC format specification:

> This is the “experiment type” flag. In most Galactic SPC files in circulation today, the `fexper` parameter has been left set at null. 

For that reason, we may want to identify the spectra so that the UI can display the right toolset (for IR, NMR, etc.)

It should be possible to have an idea of the spectra just by using `xAxisUnits`, `yAxisUnits`, and maybe `xStart`, `xEnd`. But this a bit tricky and probably needs many examples to finally accept or reject it. (the same variables are used for Atomic spectroscopy and IR sometimes, for example).

I have added a basic `guessType` function and the tests. More than identifying the spectra, it groups spectra that may use the same tool-set in the UI.

When doing it, I noticed that [properties set as `prop!` ](https://github.com/cheminfo/spc-parser/blob/54b5ad528ecd7a844bc0a9281ecd5691aa2dcf6b/src/mainHeader.ts#L7) aren't very useful for type-checking (that was my impression at least), so the code hopefully improves that aspect.

Also, I found [several files on the web](https://github.com/rohanisaac/spc) where the code seems to break (the `mainHeader.logOffset`, I think). 

I will soon add a link to a UI where it shows a large number of spectra, and the guessedType so it is easy to check whether this way is useful or not.

### CODE REVIEW

**Comment by santimirandarp** [comment]
Will split this into 2 different PRs that can be treated independently
